src/main/logutils.ts

```diff
@@ -22,10 +18,12 @@ type Metadata = {
     encounterID?: number;
     duration: number;
     result: boolean;
+    playerName: string;
 }
 
 let metadata: Metadata;
-let combatantInfoMap: Map<string, CombatantData> = new Map();
+let combatantInfoMap: Map<string, Combatant> = new Map();
+let selfCombatantInfo = new Combatant();
```

Comment:
I was thinking pure object oriented nature here. Self is not a property of a combatant. it is a description. So if we want pure object oriented this will lead down a slippery slope as to what apis we allow into pure object classes.

Reply:
Fair, good argument!